Tilers/IfcTiler/ifcObjectGeom.py

```diff
@@ -0,0 +1,297 @@
+        vertexList.append(center)            
+        vertexList.append(center + extrudVector)
+
+        for i in range(len(vertexList)) : 
+            vertexList[i] = np.dot(np.array(vertexList[i]),direction) + position
+
+        nb_points = len(points)
+        i = 0
+        for i in range(nb_points - 1) :
+            indice = i+1
+            indexList.append([indice,(nb_points * 2) + 1,indice + 1])
+            indexList.append([indice + nb_points,indice + nb_points + 1,(nb_points * 2) + 2])
+        
+        i = 0
+        for i in range(nb_points - 1) :
+            indice = i+1
+            indexList.append([indice,indice + 1,indice + nb_points ])
+            indexList.append([indice + 1,indice + nb_points + 1,indice + nb_points])
+
+        return vertexList, indexList
+
+    def getPosition(self,ObjectPlacement) :
+        listPosition = list()
+        position = np.array(ObjectPlacement.RelativePlacement.Location.Coordinates)
+        listPosition.append(position)
+        placementRelTo = ObjectPlacement.PlacementRelTo
+        
+        while (placementRelTo) :
+            if(placementRelTo.PlacesObject[0].is_a("IfcSite")):
+                break
+            listPosition.append(np.array(placementRelTo.RelativePlacement.Location.Coordinates))
+            placementRelTo = placementRelTo.PlacementRelTo
+        return listPosition
+
+    def getDirections(self,ObjectPlacement) :
+        listDirection = list()
+        axis = [0,0,1]
+        if (ObjectPlacement.RelativePlacement.Axis) :
+            axis = ObjectPlacement.RelativePlacement.Axis.DirectionRatios
+        
+        refDirection = [1,0,0]
+        if (ObjectPlacement.RelativePlacement.RefDirection) :
+            refDirection = ObjectPlacement.RelativePlacement.RefDirection.DirectionRatios 
+
+        listDirection.append(computeDirection(axis,refDirection))
+        
+        placementRelTo = ObjectPlacement.PlacementRelTo
+        
+        while (placementRelTo) :
+            if(placementRelTo.PlacesObject[0].is_a("IfcSite")):
+                break
+            axis = [0,0,1]
+            if (placementRelTo.RelativePlacement.Axis) :
+                axis = placementRelTo.RelativePlacement.Axis.DirectionRatios
+            
+            refDirection = [1,0,0]
+            if (placementRelTo.RelativePlacement.RefDirection) :
+                refDirection = placementRelTo.RelativePlacement.RefDirection.DirectionRatios 
+
+            listDirection.append(computeDirection(axis,refDirection))
+            placementRelTo = placementRelTo.PlacementRelTo
+
+        return listDirection
+
+    def getElevation(self) :
+        elevation = 0
+        if(self.ifcObject.ContainedInStructure) :
+            if(not(self.ifcObject.ContainedInStructure[0].RelatingStructure.is_a("IfcSpace"))) :
+                elevation += self.ifcObject.ContainedInStructure[0].RelatingStructure.Elevation
+
+        if(self.convertToMeter):
+            elevation /= 100
```

Comment:
Harwires the fact that the input IFC uses centimeters as unit...

Reply:
Opened [an issue for this](https://github.com/VCityTeam/py3dtiles-tilers/issues/4).